### PR TITLE
add 2 fields to reframe pattern

### PR DESCRIPTION
**Pull request by smoors**
git `commit` and `num_tasks_per_node`
see https://github.com/sisc-hpc/reframe-tests/pull/10

### CODE REVIEW

File: files/reframe [hunk 1]
```diff
@@ -1,3 +1,3 @@
-REFRAME_GENERAL username=%{USERNAME_PAT}\|version=%{DATA:version}\|name=%{DATA:name}\|system=%{DATA:system}\|partition=%{DATA:partition}\|environ=%{DATA:environ}\|num_tasks=%{INT:num_tasks:int}\|tags=%{DATA:tags}\|jobid=%{DATA:jobid}
+REFRAME_GENERAL username=%{USERNAME_PAT}\|version=%{DATA:version}\|commit=%{DATA:commit}\|name=%{DATA:name}\|system=%{DATA:system}\|partition=%{DATA:partition}\|environ=%{DATA:environ}\|num_tasks=%{INT:num_tasks:int}\|num_tasks_per_node=%{INT:num_tasks_per_node:int}\|tags=%{DATA:tags}\|jobid=%{DATA:jobid}
```
Comment (wpoely86) [inline files/reframe]: can you make it optional? So it works on the new and old at the same time?

Comment (wpoely86) [inline files/reframe]:
```suggestion
REFRAME_GENERAL username=%{USERNAME_PAT}\|version=%{DATA:version}\|(?:commit=%{DATA:commit}\|)?name=%{DATA:name}\|system=%{DATA:system}\|partition=%{DATA:partition}\|environ=%{DATA:environ}\|num_tasks=%{INT:num_tasks:int}\|num_tasks_per_node=%{INT:num_tasks_per_node:int}\|tags=%{DATA:tags}\|jobid=%{DATA:jobid}
```

Comment (wpoely86) [reply files/reframe]: and leave at least one test with the old pattern

**Review by wpoely86 — CHANGES_REQUESTED** [review]
version bump!

File: files/reframe (hunk 1 above)
Comment (smoors) [reply files/reframe]: why not just (...)? instead of (?:...)?
I tested it in ES and it gives exactly the same result

Comment (wpoely86) [reply files/reframe]: a capture vs a non-capturing group.

There is no point in capturing this group? Only the commit itself?

**Review by wpoely86 — APPROVED** [review]

**Pull request merged**